compiler/modules/front/eval.bal

```diff
@@ -286,13 +284,20 @@ function intLiteralValue(err:SemanticContext cx, t:SemType? expectedType, s:IntL
 
 // Since the binary floating point literal is parsed correctly,
 // it is impossible to return an error.
-function floatFromFpLiteral(string digits) returns float {
-    return checkpanic float:fromString(digits);
+function floatFromFpLiteral(err:SemanticContext cx, string digits, Position pos, s:IntLiteralBase base = 10) returns float|err:Semantic {
+    float|error f = base == 10 ? float:fromString(digits) : float:fromHexString("0x" + digits + "p0");
+    if f is error {
+        return cx.semanticErr("invalid float literal", cause=f, pos=pos);
+    }
+    return f;
 }
 
 // Even if the decimal floating point literal is parsed correctly,
 // overflows should return an error.
-function decimalFromFpLiteral(err:SemanticContext cx, string decimalStr, Position pos) returns decimal|err:Semantic {
+function decimalFromFpLiteral(err:SemanticContext cx, string decimalStr, Position pos, s:IntLiteralBase base = 10) returns decimal|err:Semantic|err:Unimplemented {
+    if base == 16 {
+        return cx.unimplementedErr("hexadecimal int cannot be converted to decimal", pos=pos);
```

Comment:
Make this a semantic err, and document in subset13 that we have implemented the proposed spec change that I mentioned.